### PR TITLE
Add Service Catalog Feature Back-end Implementation

### PR DESCRIPTION
**Pull request by YasasRangika**
### Objective
The main purpose of this Service Catalog feature is to create a connection between API Management and Integration layers by making integration services discoverable to API Management layer.

### Approach
Service catalog portal will consist of service catalog web user interface which is available in the Publisher web portal, REST API, and data persistent components. Each tenant will have a separate service catalog space, so that only the users from the same tenant space can read from and write to the catalog.
For this release it will available four main endpoints that can use to work with Service Catalog.

**POST /service-entries/import**

- Import a service by providing an archived service
- This endpoint currently support with overwrite parameter = true
- If the client send request with payload zip that contains directories relevant to each service with overwrite=true, then this endpoint will overwrite all the entries.

**GET /service-entries**

- Retrieve or search services in the service catalog of the user's organization or tenant.
- Client should send the request with shrink=true

**DELETE ​/service-entries​/{serviceKey}**

- Delete a service by providing the service key

**GET /services/export**

- Export a service as an archived zip file.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/model/ExportArchive.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.wso2.carbon.apimgt.rest.api.service.catalog.model;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/model/ExportArchive.java:1]: missing License

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/impl/ServiceEntriesApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class ServiceEntriesApiServiceImpl implements ServiceEntriesApiService {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/impl/ServiceEntriesApiServiceImpl.java:55]: Add class comment

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+ */
+public class ServiceCatalogDAO {
+
+    private static final Log log = LogFactory.getLog(ServiceCatalogDAO.class);
+    private static ServiceCatalogDAO INSTANCE = null;
+
+    /**
+     * Method to get the instance of the ServiceCatalogDAO.
+     *
+     * @return {@link ApiMgtDAO} instance
```
Comment (vithu30) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java:46]: Update the correct class name

File: features/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog.feature/pom.xml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.wso2.carbon.apimgt</groupId>
+        <artifactId>apimgt-feature</artifactId>
+        <version>6.8.158-SNAPSHOT</version>
```
Comment (vithu30) [inline features/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog.feature/pom.xml:24]: Shall we update the correct version?

Comment (vithu30) [inline features/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog.feature/pom.xml:24]: Shall we update the correct version

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/model/ExportArchive.java (hunk 1 above)
Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/model/ExportArchive.java:19]: Added in 4208bf9c0f5c799e3876e177032977fb9c64e85f

**Review by mushthaq33 — APPROVED** [review]

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**